src/spi_software.c

```diff
@@ -29,31 +34,52 @@ command_spi_set_software_bus(uint32_t *args)
     ss->mosi = gpio_out_setup(args[2], 0);
     ss->sclk = gpio_out_setup(args[3], 0);
     ss->mode = mode;
+    ss->sck_ticks = pulse_ticks;
     spidev_set_software_bus(spi, ss);
 }
 DECL_COMMAND(command_spi_set_software_bus,
-             "spi_set_software_bus oid=%c miso_pin=%u mosi_pin=%u sclk_pin=%u"
-             " mode=%u rate=%u");
+             "spi_set_sw_bus oid=%c miso_pin=%u mosi_pin=%u sclk_pin=%u"
+             " mode=%u pulse_ticks=%u");
 
 void
 spi_software_prepare(struct spi_software *ss)
 {
     gpio_out_write(ss->sclk, ss->mode & 0x02);
 }
 
+static void
+spi_delay(uint32_t *time, uint32_t ticks)
+{
+    uint32_t cur = timer_read_time();
+    *time += ticks;
+    // Handle interrupts
+    if (timer_is_before(*time, cur)) {
+        *time = cur;
+        return;
+    }
+    while (timer_is_before(cur, *time))
+        cur = timer_read_time();
+    *time = cur;
+}
+
 void
 spi_software_transfer(struct spi_software *ss, uint8_t receive_data
                       , uint8_t len, uint8_t *data)
 {
+    uint32_t t1 = ss->sck_ticks >> 1;
+    uint32_t t2 = t1 + (ss->sck_ticks & 1);
```

Comment:
Super minor, but I think this may be a little more readable as: `uint32_t t2 = ss->sck_ticks - t1;`